### ISSUE TITLE
Attention maps

### ISSUE THREAD

**Issue by Tato14** [lucidrains/vit-pytorch#1]
Hi! First, thanks for the great resource. 
I was wondering how difficult would be to implement the attention results they show in the Fig. 6 and Fig 13 of the paper.
I am not quite familiar with transformers. This is similar to GradCam o some different approach?

**Comment by lucidrains** [comment]
@Tato14 Hi Joan! Seems like the approach came from https://arxiv.org/pdf/2005.00928.pdf I'll have to read it after I get through my queue of papers this week to see how difficult it is to implement! Feel free to keep this issue open in the meanwhile

**Comment by lucidrains** [comment]
@Tato14 the naive attention map for individual layers is this variable `attn` https://github.com/lucidrains/vit-pytorch/blob/main/vit_pytorch/vit_pytorch.py#L45

**Comment by lukasfolle** [comment]
Its probably https://github.com/lucidrains/vit-pytorch/blob/6c8dfc185ea41f4d2388e4d33bbb76f900ff8a0a/vit_pytorch/vit_pytorch.py#L63.

**Comment by PascalHbr** [comment]
Why is the softmax only applied to dim=-1? Shouldn't the softmax be calculated over the last 2 dimensions?

**Comment by suarezjessie** [comment]
Hi @lucidrains, is there any news for the attention map visualization? Thanks!

**Comment by jpgard** [comment]
It seems this has been implemented; see the description in the README [here](https://github.com/lucidrains/vit-pytorch#accessing-attention).

However, I'm noticing some strange behavior -- when I train a model which achieves good performance on a task, and pull the attention weights, only the weights for Larger 0 are nonzero -- the attention maps for all other layers and heads in my trained model are all constant, with value 0.02. Could this be a bug? Otherwise, model train/test performance is fine. Example attention weights for a single input shown below.

![image](https://user-images.githubusercontent.com/7265452/114252066-578d9380-9958-11eb-8e3c-58a04c403891.png)
